### PR TITLE
Add snowflake o auth example

### PR DESCRIPTION
**Pull request by wangxinlei**
**Things To Remember:** 

1. Be sure to merge into the right branch. The choices are:
* **`master`** if your PR runs on the current version of Tableau.
* **`dev-2020.1`**
* **`2020.2`**
2. You should update at least one of the following:
* The changelog of the project you modified.
* The root `README.md` if the PR is a release of the SDK, Connector Packager, or TDVT.  
=====

### CODE REVIEW

File: samples/scenarios/snowflake_oauth/connectionBuilder.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+(function dsbuilder(attr)
+{
+    var params = {};
+    var authAttrValue = attr[connectionHelper.attributeAuthentication];
+
+    for (var i = 0, keys = Object.keys(attr), ii = keys.length; i < ii; i++) {
+          logging.log(keys[i] + '|' + attr[keys[i]]);
+    }
```
Comment (pvanderknyff) [inline samples/scenarios/snowflake_oauth/connectionBuilder.js:8]: We probably don't want to log these.

File: samples/scenarios/snowflake_oauth/oauthConfig.xml [hunk 2]
```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0" encoding="utf-8"?>
+<pluginOAuthConfig>	 
+	<dbclass>snowflake_oauth</dbclass>
+    <clientIdDesktop>UP+iCpaEoSG9qM1Q1+Xx6kpxwF0=</clientIdDesktop>
+    <clientSecretDesktop>EAofGQyN1du/vqpIK/eThPGhLNBHNyb9DA9Js8bk2YA=</clientSecretDesktop>
```
Comment (pvanderknyff) [inline samples/scenarios/snowflake_oauth/oauthConfig.xml:5]: Are we comfortable with these being out in the open? If there's any doubt, we should just replace these fields with XXXXXXX, the connector doesn't technically need to work, just serve as an example.

**Review by pvanderknyff — COMMENTED** [review]
Please merge into dev-2021.1 branch. Master always targets the latest released version, which is 2020.4 right now, and this feature won't be released until 2021.1.

File: samples/scenarios/snowflake_oauth/oauthConfig.xml (hunk 2 above)
Comment (wangxinlei) [reply samples/scenarios/snowflake_oauth/oauthConfig.xml:5]: Will mask them

File: samples/scenarios/snowflake_oauth/connectionBuilder.js (hunk 1 above)
Comment (wangxinlei) [reply samples/scenarios/snowflake_oauth/connectionBuilder.js:8]: Will remove

**Comment by wangxinlei** [comment]
> Please merge into dev-2021.1 branch. Master always targets the latest released version, which is 2020.4 right now, and this feature won't be released until 2021.1.

OK I'll close this one and open another one to dev-2021.1.

**Pull request closed without merging**